### PR TITLE
chore: added GH app token in add-to-project workflow

### PR DESCRIPTION
**Pull request by drodriguez-305**
### Which problem does the PR fix?

https://github.com/camunda/distribution/issues/161

### What's in this PR?
replace PAT tokens with GH app tokens generated in GHA. 

### Checklist

Please make sure to follow our [Contributing Guide](../blob/main/CONTRIBUTING.md).

**Before opening the PR:**

- [ ] In the repo's root dir, run `make go.update-golden-only`.
- [ ] There is no other open [pull request](../pulls) for the same update/change.
- [ ] Tests for charts are added (if needed).
- [ ] In-repo [documentation](../blob/main/CONTRIBUTING.md#documentation) are updated (if needed).

**After opening the PR:**

- [x] Did you sign our CLA (Contributor License Agreement)? It will show once you open the PR.
- [ ] Did all checks/tests pass in the PR?

### CODE REVIEW

File: .github/workflows/add-to-project.yml [hunk 1]
```diff
@@ -11,7 +11,14 @@ jobs:
...
+      - name: Generate GitHub token
+        uses: tibdex/github-app-token@v2
+        id: generate-github-token
+        with:
+          app_id: ${{ secrets.GH_APP_ID }}
+          private_key: ${{ secrets.GH_APP_PRIVATE_KEY }}
+      - name: Add to project
+        uses: actions/add-to-project@v0.5.0
         with:
           project-url: https://github.com/orgs/camunda/projects/33
```
Comment (aabouzaid) [inline .github/workflows/add-to-project.yml:23]: Let's take the chance and set that as a GHA env var.

Comment (drodriguez-305) [reply .github/workflows/add-to-project.yml:26]: done.

**Review by jessesimpson36 — APPROVED** [review]

File: .github/workflows/add-to-project.yml [hunk 2]
```diff
@@ -5,13 +5,23 @@ on:
     types:
       - opened
       - transferred
+env:
```
Comment (aabouzaid) [inline .github/workflows/add-to-project.yml:8]: The same comment here about the vars from the other PR.

Comment (drodriguez-305) [reply .github/workflows/add-to-project.yml]: Completed, I had left it thinking we might use it in the future. We should be all set now.

**Review by aabouzaid — APPROVED** [review]
LGTM, thank you :+1:

**Pull request merged**